src/examples/threading.jl

```diff
@@ -1,18 +1,18 @@
 """
-The threading macro is like a more flexible version of the `|>` operator.
+The threading macro is like a more flexible version of the [`|>`](@ref) operator.
```

Comment:
Most of those documentations don't appear in MacroTools docs: https://github.com/MikeInnes/MacroTools.jl/blob/master/docs/src/utilities.md#other-utilities

So I just assumed the usages within REPL and Juno's documentation pane.
But I don't think `@ref` is so important, so I can remove them if you want.

Reply:
What I mean is that while `|>` does have docs, they don't come from MacroTools but from Base Julia. My suspicion is that this will mean that when we build the MacroTools docs, this ends up as a broken link (since Documenter may not know how to link to Base Julia docs). I'm not sure about that so it would be great to confirm either way.

Reply:
ah got it ! Will check sooner and get you back.

Reply:
Documenter even doesn't try to make docs of things that don't appear in the documentation source, so it just ignores those problematic links:

> julia --project=docs/ docs/make.jl
```
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: ExpandTemplates: expanding markdown templates.
[ Info: CrossReferences: building cross-references.
[ Info: CheckDocument: running document checks.
[ Info: Populate: populating indices.
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
┌ Warning: Documenter could not auto-detect the building environment Skipping deployment.
└ @ Documenter C:\Users\aviat\.julia\packages\Documenter\O67pl\src\deployconfig.jl:41
```

Reply:
So what happens to the link in the final output, then?

Reply:
Those documentations themselves (e.g. a documentation for `@>>`) don't appear in the final output, as they don't in [the current documentation](https://mikeinnes.github.io/MacroTools.jl/stable/).

They would only appear in REPL, Juno's documentation, etc.

Reply:
Ok, I didn't realise you actually weren't adding these to the docs. Even then it's still relevant since we may want to do that at some point.

My question stands though. What happens to the ``[`|>`](@ref)`` link in the repl or Juno? Does the link work, or is it broken?

Reply:
In Juno, those `@ref`s work since DocSeeker can search documentations in all the loaded modules (even though it just relies on its custom scoring strategy).
In REPL, those `@ref` doesn't do AFAIU.

> Even then it's still relevant since we may want to do that at some point.

okay, so how about just removing links ? I'm okay with that after all :)